### ISSUE TITLE
feat(web): retain scroll position in details view when selecting another zone/country

### ISSUE THREAD

**Issue by consideRatio** [electricitymaps/electricitymaps-contrib#7618]
**Is your feature request related to a problem? Please describe.**

When looking at for example the net exchange chart in the details panel for a zone, and select another zone/country, for example having selected Sweden and then clicking on Denmark, the scroll position in the details view gets reset. I think this isn't ideal, and that it would be better to retain some kind of scroll position to help users retain focus on what they are interested in.

As a practical example, I'd like to look at various charts that I need to scroll down to see, but when selecting another zone/country to compare with, I prefer to not need to scroll back down in between doing it.

**Describe the solution you'd like**

I think the scroll position should be retained and restored when loading the zone/country data. Exactly how, I'm not sure.

**Describe solution implementations you've considered**

- To always reset the scroll position (current behavior)
- To restore the scroll position to the same relative position (may be good enough, probably easy)
- To restore the scroll position to the content relative position (may be more complicated, but perhaps more on point?)

**Additional context**

Initially observed by @IngvarSundell. A sibling issue to this is #7617.

**Comment by razyneko** [comment]
hey @consideRatio , considering the issue you mean if we are currently looking at hourly net exchange of some country so after clicking on another country scroll should be exact at hourly net exchange for that country ?

**Comment by consideRatio** [comment]
Yes ideally, but its not obvious to me its possible if for example the total content height also changes and/or various graphs start/stop showing between changes.

However, currently the scroll position resets to the top all the time - and that can likely be improved on I think. I'd say that the goal may be that the new scroll position show as much of the previous content also for the new country selected.

**Comment by razyneko** [comment]
@consideRatio yeah i saw some countries were missing some data and some even don't have any, this is gonna be a good one, i have forked and cloned the repo, lemme try at least

**Comment by DanNguyen8189** [comment]
Hey! I saw this was open for a while and thought I'd give it a shot - I opened a pull request here: https://github.com/electricitymaps/electricitymaps-contrib/pull/8547

Right now, it's set so that it'll scroll to the top if the current zone is missing the data the previous zone was scrolled to
Explanation of changes are under the description - Let me know if there's something missing or should be changed

**Comment by madsnedergaard** [comment]
Thanks for raising this 🙌 
It's a good idea, we'll keep it in mind as part of a larger map UI revamp.
We're going to track it internally as we're focusing the GitHub repository more on the data-side, so I will close this issue for now  :)
